The go module `github.com/sirupsen/logrus` does not support loongson3(mips64le) and loongarch64(loong64) and can't be built on riscv64 and ppc64el. This may cause dae logging related functions to be abnormal.

This build information can also be found in dae's github action.

<details>
  <summary>supported targets:</summary>

```
go: downloading github.com/sirupsen/logrus v1.9.3
Supported targets:
	bpf
	bpfel
	bpfeb
	386
	amd64
	arm
	arm64
	arm64be
	armbe
	ppc64
	ppc64le
	riscv64
	s390
	s390x
	sparc
	sparc64

Error: "mips64le": unsupported target
exit status 1
```

</details>


<details>
  <summary>built on riscv64</summary>

```
go: downloading github.com/sirupsen/logrus v1.9.3
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:137:23: error: no member named 'a1' in 'struct user_regs_struct'
  137 |         meta->second_param = PT_REGS_PARM2(ctx);
      |                              ^~~~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:486:46: note: expanded from macro 'PT_REGS_PARM2'
  486 | #define PT_REGS_PARM2(x) (__PT_REGS_CAST(x)->__PT_PARM2_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:357:24: note: expanded from macro '__PT_PARM2_REG'
  357 | #define __PT_PARM2_REG a1
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:227:1: error: no member named 'a0' in 'struct user_regs_struct'
  227 | KPROBE_SKB_AT(1)
      | ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
  223 |     struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx);      \
      |                                              ^~~~~~~~~~~~~~~~~~~~
<scratch space>:169:1: note: expanded from here
  169 | PT_REGS_PARM1
      | ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:485:46: note: expanded from macro 'PT_REGS_PARM1'
  485 | #define PT_REGS_PARM1(x) (__PT_REGS_CAST(x)->__PT_PARM1_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:356:24: note: expanded from macro '__PT_PARM1_REG'
  356 | #define __PT_PARM1_REG a0
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:228:1: error: no member named 'a1' in 'struct user_regs_struct'
  228 | KPROBE_SKB_AT(2)
      | ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
  223 |     struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx);      \
      |                                              ^~~~~~~~~~~~~~~~~~~~
<scratch space>:178:1: note: expanded from here
  178 | PT_REGS_PARM2
      | ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:486:46: note: expanded from macro 'PT_REGS_PARM2'
  486 | #define PT_REGS_PARM2(x) (__PT_REGS_CAST(x)->__PT_PARM2_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:357:24: note: expanded from macro '__PT_PARM2_REG'
  357 | #define __PT_PARM2_REG a1
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:229:1: error: no member named 'a2' in 'struct user_regs_struct'
  229 | KPROBE_SKB_AT(3)
      | ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
  223 |     struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx);      \
      |                                              ^~~~~~~~~~~~~~~~~~~~
<scratch space>:187:1: note: expanded from here
  187 | PT_REGS_PARM3
      | ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:487:46: note: expanded from macro 'PT_REGS_PARM3'
  487 | #define PT_REGS_PARM3(x) (__PT_REGS_CAST(x)->__PT_PARM3_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:358:24: note: expanded from macro '__PT_PARM3_REG'
  358 | #define __PT_PARM3_REG a2
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:230:1: error: no member named 'a3' in 'struct user_regs_struct'
  230 | KPROBE_SKB_AT(4)
      | ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
  223 |     struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx);      \
      |                                              ^~~~~~~~~~~~~~~~~~~~
<scratch space>:196:1: note: expanded from here
  196 | PT_REGS_PARM4
      | ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:488:46: note: expanded from macro 'PT_REGS_PARM4'
  488 | #define PT_REGS_PARM4(x) (__PT_REGS_CAST(x)->__PT_PARM4_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:359:24: note: expanded from macro '__PT_PARM4_REG'
  359 | #define __PT_PARM4_REG a3
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:231:1: error: no member named 'a4' in 'struct user_regs_struct'
  231 | KPROBE_SKB_AT(5)
      | ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
  223 |     struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx);      \
      |                                              ^~~~~~~~~~~~~~~~~~~~
<scratch space>:205:1: note: expanded from here
  205 | PT_REGS_PARM5
      | ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:489:46: note: expanded from macro 'PT_REGS_PARM5'
  489 | #define PT_REGS_PARM5(x) (__PT_REGS_CAST(x)->__PT_PARM5_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:360:24: note: expanded from macro '__PT_PARM5_REG'
  360 | #define __PT_PARM5_REG a4
      |                        ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/trace.c:236:18: error: no member named 'a0' in 'struct user_regs_struct'
  236 |         u64 skb = (u64) PT_REGS_PARM1(ctx);
      |                         ^~~~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:485:46: note: expanded from macro 'PT_REGS_PARM1'
  485 | #define PT_REGS_PARM1(x) (__PT_REGS_CAST(x)->__PT_PARM1_REG)
      |                           ~~~~~~~~~~~~~~~~~  ^
/var/cache/acbs/build/acbs.4uyem51s/dae/trace/kern/headers/bpf_tracing.h:356:24: note: expanded from macro '__PT_PARM1_REG'
  356 | #define __PT_PARM1_REG a0
      |                        ^
7 errors generated.
Error: can't execute clang: exit status 1
exit status 1
```

</details>


<details>
  <summary>built on ppc64el</summary>

```
go: downloading github.com/sirupsen/logrus v1.9.3
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:137:23: error: no member named 'gpr' in 'struct pt_regs'
137 | meta->second_param = PT_REGS_PARM2(ctx);
| ^~~~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:486:46: note: expanded from macro 'PT_REGS_PARM2'
486 | #define PT_REGS_PARM2(x) (__PT_REGS_CAST(x)->__PT_PARM2_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:291:24: note: expanded from macro '__PT_PARM2_REG'
291 | #define __PT_PARM2_REG gpr[4]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:227:1: error: no member named 'gpr' in 'struct pt_regs'
227 | KPROBE_SKB_AT(1)
| ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
223 | struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx); \
| ^~~~~~~~~~~~~~~~~~~~
<scratch space>:169:1: note: expanded from here
169 | PT_REGS_PARM1
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:485:46: note: expanded from macro 'PT_REGS_PARM1'
485 | #define PT_REGS_PARM1(x) (__PT_REGS_CAST(x)->__PT_PARM1_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:290:24: note: expanded from macro '__PT_PARM1_REG'
290 | #define __PT_PARM1_REG gpr[3]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:228:1: error: no member named 'gpr' in 'struct pt_regs'
228 | KPROBE_SKB_AT(2)
| ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
223 | struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx); \
| ^~~~~~~~~~~~~~~~~~~~
<scratch space>:178:1: note: expanded from here
178 | PT_REGS_PARM2
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:486:46: note: expanded from macro 'PT_REGS_PARM2'
486 | #define PT_REGS_PARM2(x) (__PT_REGS_CAST(x)->__PT_PARM2_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:291:24: note: expanded from macro '__PT_PARM2_REG'
291 | #define __PT_PARM2_REG gpr[4]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:229:1: error: no member named 'gpr' in 'struct pt_regs'
229 | KPROBE_SKB_AT(3)
| ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
223 | struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx); \
| ^~~~~~~~~~~~~~~~~~~~
<scratch space>:187:1: note: expanded from here
187 | PT_REGS_PARM3
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:487:46: note: expanded from macro 'PT_REGS_PARM3'
487 | #define PT_REGS_PARM3(x) (__PT_REGS_CAST(x)->__PT_PARM3_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:292:24: note: expanded from macro '__PT_PARM3_REG'
292 | #define __PT_PARM3_REG gpr[5]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:230:1: error: no member named 'gpr' in 'struct pt_regs'
230 | KPROBE_SKB_AT(4)
| ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
223 | struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx); \
| ^~~~~~~~~~~~~~~~~~~~
<scratch space>:196:1: note: expanded from here
196 | PT_REGS_PARM4
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:488:46: note: expanded from macro 'PT_REGS_PARM4'
488 | #define PT_REGS_PARM4(x) (__PT_REGS_CAST(x)->__PT_PARM4_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:293:24: note: expanded from macro '__PT_PARM4_REG'
293 | #define __PT_PARM4_REG gpr[6]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:231:1: error: no member named 'gpr' in 'struct pt_regs'
231 | KPROBE_SKB_AT(5)
| ^~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:223:46: note: expanded from macro 'KPROBE_SKB_AT'
223 | struct sk_buff *skb = (struct sk_buff *) PT_REGS_PARM##X(ctx); \
| ^~~~~~~~~~~~~~~~~~~~
<scratch space>:205:1: note: expanded from here
205 | PT_REGS_PARM5
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:489:46: note: expanded from macro 'PT_REGS_PARM5'
489 | #define PT_REGS_PARM5(x) (__PT_REGS_CAST(x)->__PT_PARM5_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:294:24: note: expanded from macro '__PT_PARM5_REG'
294 | #define __PT_PARM5_REG gpr[7]
| ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/trace.c:236:18: error: no member named 'gpr' in 'struct pt_regs'
236 | u64 skb = (u64) PT_REGS_PARM1(ctx);
| ^~~~~~~~~~~~~~~~~~
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:485:46: note: expanded from macro 'PT_REGS_PARM1'
485 | #define PT_REGS_PARM1(x) (__PT_REGS_CAST(x)->__PT_PARM1_REG)
| ~~~~~~~~~~~~~~~~~ ^
/var/cache/acbs/build/acbs.i6owkpq5/daed/wing/dae-core/trace/kern/headers/bpf_tracing.h:290:24: note: expanded from macro '__PT_PARM1_REG'
290 | #define __PT_PARM1_REG gpr[3]
| ^
7 errors generated.
Error: can't execute clang: exit status 1
exit status 1 
```

</details>

--- a/app-network/dae/autobuild/build
+++ b/app-network/dae/autobuild/build
@@ -1,0 +1,23 @@
+abinfo "Building dae ..."
+make VERSION="$PKGVER"
+
+abinfo "Installing dae ..."
+install -Dvm755 "$SRCDIR"/dae -t "$PKGDIR"/usr/bin
+install -Dvm644 "$SRCDIR"/install/dae.service -t "$PKGDIR"/usr/lib/systemd/system
+# Should be set to 640 or 600.
+# See: https://github.com/daeuniverse/dae/blob/d3ab0b25e705399d7e610d6f09453623674e20d3/common/subscription/subscription.go#L119
+install -Dvm640 "$SRCDIR"/install/empty.dae "$PKGDIR"/etc/dae/config.dae
+install -Dvm644 "$SRCDIR"/example.dae "$PKGDIR"/etc/dae/config.dae.example
+
+abinfo "Linking to geofile directory ..."
+install -dv "$PKGDIR"/usr/share/dae
+ln -sv /usr/share/v2ray-rules-dat/{geoip.dat,geosite.dat} \
+    "$PKGDIR"/usr/share/dae
+
+abinfo "Installing completion files ..."
+install -Dvm644 "$SRCDIR"/install/shell-completion/dae.bash \
+    "$PKGDIR"/usr/share/bash-completion/completions/dae
+install -Dvm644 "$SRCDIR"/install/shell-completion/dae.zsh \
+    "$PKGDIR"/usr/share/zsh/site-functions/_dae
+install -Dvm644 "$SRCDIR"/install/shell-completion/dae.fish \
+    "$PKGDIR"/usr/share/fish/vendor_completions.d/dae.fish

--- a/app-network/dae/autobuild/defines
+++ b/app-network/dae/autobuild/defines
@@ -1,0 +1,13 @@
+PKGNAME="dae"
+PKGSEC=net
+PKGDES="Transparent proxy solution based on eBPF"
+PKGDEP="gcc-runtime v2ray-rules-dat"
+PKGCONFL="daed"
+BUILDDEP="go llvm"
+
+# FIXME: Autobuild does not yet support splitting debug symbols from Go executables.
+ABSPLITDBG=0
+# Go module "github.com/sirupsen/logrus" does not support loongson3 and loongarch64
+# and build error on ppc64el and riscv64.
+# See: https://github.com/daeuniverse/dae/issues/696
+FAIL_ARCH="!(amd64|arm64)"

--- a/app-network/dae/autobuild/prepare
+++ b/app-network/dae/autobuild/prepare
@@ -1,0 +1,7 @@
+abinfo "Setting GOFLAGS ..."
+export GOFLAGS="-trimpath -mod=readonly -modcacherw -buildmode=pie"
+
+abinfo "Setting other environment variables ..."
+export BUILD_ARGS="${GOFLAGS}"
+export CFLAGS="-fno-stack-protector"
+export CGO_ENABLED=1

--- a/app-network/dae/spec
+++ b/app-network/dae/spec
@@ -1,0 +1,4 @@
+VER=0.8.0
+SRCS="git::commit=v$VER;copy-repo=true::https://github.com/daeuniverse/dae.git"
+CHKSUMS="SKIP"
+CHKUPDATE="anitya::id=369479"

--- a/app-network/daed/autobuild/build
+++ b/app-network/daed/autobuild/build
@@ -1,0 +1,17 @@
+abinfo "Building daed ..."
+make VERSION="$PKGVER"
+
+abinfo "Installing daed ..."
+install -Dvm755 "$SRCDIR"/daed -t "$PKGDIR"/usr/bin
+install -Dvm644 "$SRCDIR"/install/daed.service \
+    -t "$PKGDIR"/usr/lib/systemd/system
+install -Dvm644 "$SRCDIR"/install/daed.desktop \
+    -t "$PKGDIR"/usr/share/applications
+install -Dvm644 "$SRCDIR"/install/icons/512x512.png \
+    "$PKGDIR"/usr/share/icons/hicolor/512x512/apps/daed.png
+install -dv "$PKGDIR"/etc/daed
+
+abinfo "Linking to geofile directory ..."
+install -dv "$PKGDIR"/usr/share/daed
+ln -sv /usr/share/v2ray-rules-dat/{geoip.dat,geosite.dat} \
+    "$PKGDIR"/usr/share/daed

--- a/app-network/daed/autobuild/defines
+++ b/app-network/daed/autobuild/defines
@@ -1,0 +1,13 @@
+PKGNAME="daed"
+PKGSEC=net
+PKGDES="Transparent proxy solution based on eBPF, bundled with dashboard"
+PKGDEP="gcc-runtime v2ray-rules-dat"
+PKGCONFL="dae"
+BUILDDEP="go llvm nodejs"
+
+# FIXME: Autobuild does not yet support splitting debug symbols from Go executables.
+ABSPLITDBG=0
+# Go module "github.com/sirupsen/logrus" does not support loongson3 and loongarch64
+# and build error on ppc64el and riscv64.
+# See: https://github.com/daeuniverse/dae/issues/696
+FAIL_ARCH="!(amd64|arm64)"

--- a/app-network/daed/autobuild/prepare
+++ b/app-network/daed/autobuild/prepare
@@ -1,0 +1,6 @@
+abinfo "Setting GOFLAGS ..."
+export GOFLAGS="-trimpath -mod=readonly -modcacherw -buildmode=pie -ldflags=-linkmode=external"
+
+abinfo "Setting other environment variables ..."
+export CFLAGS="-fno-stack-protector"
+export CGO_ENABLED=1

--- a/app-network/daed/spec
+++ b/app-network/daed/spec
@@ -1,0 +1,4 @@
+VER=0.8.0
+SRCS="git::commit=v$VER;copy-repo=true::https://github.com/daeuniverse/daed.git"
+CHKSUMS="SKIP"
+CHKUPDATE="anitya::id=375373"

